### ISSUE TITLE
Incorrect namespace is being injected into workspace pods

### ISSUE THREAD

**Issue by kibblerz** [eclipse-che/che#19578]
### Describe the bug
Recently created a Che deployment in Kubernetes and while troubleshooting certain workspace issues, I came an error in the theia output saying the service account didn't have access to the namespace "undefined". I granted full admin access to the service account for my workspace, and it now gives this error:

Extension-Host:Promise rejection not handled in one second: Error: Cannot read Pod information. Status code: 404. Error: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"pods \"workspacee2x8iaenns1cill9.che-workspace-pod-74bd7554b-dbns8\" not found","reason":"NotFound","details":{"name":"workspacee2x8iaenns1cill9.che-workspace-pod-74bd7554b-dbns8","kind":"pods"},"code":404}
 , reason: Error: Cannot read Pod information. Status code: 404. Error: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"pods \"workspacee2x8iaenns1cill9.che-workspace-pod-74bd7554b-dbns8\" not found","reason":"NotFound","details":{"name":"workspacee2x8iaenns1cill9.che-workspace-pod-74bd7554b-dbns8","kind":"pods"},"code":404}
Extension-Host:With stack trace: Error: Cannot read Pod information. Status code: 404. Error: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"pods \"workspacee2x8iaenns1cill9.che-workspace-pod-74bd7554b-dbns8\" not found","reason":"NotFound","details":{"name":"workspacee2x8iaenns1cill9.che-workspace-pod-74bd7554b-dbns8","kind":"pods"},"code":404}

When I look at the Environment variables that are being injected into the workspace, I found that the "CHE_WORKSPACE_NAMESPACE" variable is simply set to the username (instead of <usernamce>-che). This seems to be preventing any Kubernetes API calls from succeeding.

### Che version
 - [7.28.2] latest

### Steps to reproduce

1. Install a multi-user setup on 7.28.2 via the Operator crds
2. set namespace names to '<username>-che'
3. Create a new user and a workspace
4. Examine output logs for Kubernetes API errors

### Expected behavior
The workspace pods should use the correct namespace when querying the Kubernetes API

### Runtime
   - 1.18 kubernetes
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.0", GitCommit:"af46c47ce925f4c4ad5cc8d1fca46c7b77d13b38", GitTreeState:"clean", BuildDate:"2020-12-08T17:59:43Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.9-eks-d1db3c", GitCommit:"d1db3c46e55f95d6a7d3e5578689371318f95ff9", GitTreeState:"clean", BuildDate:"2020-10-20T22:18:07Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}

### Installation method
   - [] chectl
     * chectl server:deploy --installer=operator --platform=k8s --che-operator-cr-yaml=operator.yaml --multiuser
     * chectl/7.28.2 linux-x64 node-v12.22.0

### Environment
   - [ ] my computer
       - [ ] macOS
   - [ ] Cloud
       - [ ] Amazon

**Comment by kibblerz** [comment]
After some research, I found out that apparently this "Namespace" variable that's injected isn't related to kubernetes namespaces. I fixed the error by adding an attribute for infrastructureNamespace into the devfile

**Comment by skabashnyuk** [comment]
@kibblerz can you please clarify the steps to reproduce and expectations? I do not fully understand the use case here.

**Comment by kibblerz** [comment]
When I create a new workspace in Eclipse Che, The resource monitor errors out saying it cannot access the namespace of "undefined". This is with the proper RBAC permissions applied.

I resolved this by adding the infrastructureNamespace attribute to the devfile for each workspace, specifying the workspaces' namespace. For some reason, Che isn't passing the attribute to Theia automatically.

As far as steps to reproduce, all I do to reproduce this is create a new workspace. We're currently on the latest stable version, and our setup is pretty basic. 

The only thing that may deviate from other traditional setups is that we use Istio for our cluster. To handle Che routing, I created an Nginx controller as a NodePort and routed all traffic for the allocated subdomain to the controller with an istio virtual service. Everything else in our setup is fairly typical.
